presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -363,9 +364,12 @@ private DistributedQueryRunner(
         prestoClients = prestoClientsBuilder.build();
 
         long start = nanoTime();
-        while (!allNodesGloballyVisible()) {
-            Assertions.assertLessThan(nanosSince(start), new Duration(100, SECONDS));
-            MILLISECONDS.sleep(10);
+        try {
+            waitForAllNodesGloballyVisible(start);
```

Comment:
Couldn't you just get the `start` time inside the function?

Reply:
Yes, moved it.